psk_conn.go

```diff
@@ -31,11 +32,8 @@ type pskConn struct {
 func (c *pskConn) Read(out []byte) (int, error) {
 	if c.readS20 == nil {
 		nonce := make([]byte, 24)
-		n, err := c.Conn.Read(nonce)
+		_, err := io.ReadFull(c.Conn, nonce)
 		if err != nil {
-			return 0, err
-		}
-		if n != 24 {
 			return 0, errShortNonce
```

Comment:
err is just io.ErrUnexpectedEOF (or io.EOF if zero bytes were read), so I think it makes more to return the more specific error here.
We could concat the error strings if we want to pass the exact error upfront logging.